Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -132,7 +132,14 @@ def extract
       loop do
         rev = rev.nil? ? "HEAD" : "#{rev}~1"
         rev, (path,) = Git.last_revision_commit_of_files(repo, pattern, before_commit: rev)
-        odie "Could not find #{name}! The formula or version may not have existed." if rev.nil?
+        if rev.nil? && source_tap.shallow?
+          opoo <<~EOS
+            #{source_tap} is shallow clone! Retry after running:
+              git -C "$(#{source_tap.path})" fetch --unshallow
+          EOS
+        elsif rev.nil?
+          odie "Could not find #{name}! The formula or version may not have existed."
+        end
```

Comment:
```suggestion
        if rev.nil? && source_tap.shallow?
          odie <<~EOS
            Could not find #{name} but #{source_tap} is a shallow clone! 
            Try again after running:
              git -C "$(#{source_tap.path})" fetch --unshallow
          EOS
        elsif rev.nil?
          odie "Could not find #{name}! The formula or version may not have existed."
        end
```

Reply:
Whoops, noticed another tweak: this should still also `odie` and say it wasn't found!